### DISCUSSION TITLE
Bug: People's section is empty

### DISCUSSION THREAD

**Discussion by Angelmmiguel** [photoprism/photoprism#5362] [Q&A]
**What does not work as expected?**

When accessing `/people`,  I don't see any person there. However, I already tagged 35 people manually. Even after tagging a new person on `/people/new`, they still don't appear in the `/people`. 

Checking the database, I found many columns with `NULL` values in the `subjects` table (See the example files section).

**How can we reproduce it?**

1. Go to ` /people/new`
2. Tag a new person
3. Go to `/people`
4. See an empty page and a toast message saying "There's no people"

**What behavior do you expect?**

I expect to see the people I already tagged manually.

**Can you provide us with example files for testing or screenshots?**

Checking the database, I can see data in the `subjects` table. However, many columns are empty:

```
MariaDB [photoprism]> Select * FROM subjects LIMIT 2;
+------------------+-----------+----------+-----------+-----------+------------+----------+------------+---------------+-------------+--------------+---------------+------------+-------------+-------+-----------+---------------+---------------------+---------------------+------------+
| subj_uid         | subj_type | subj_src | subj_slug | subj_name | subj_alias | subj_bio | subj_notes | subj_favorite | subj_hidden | subj_private | subj_excluded | file_count | photo_count | thumb | thumb_src | metadata_json | created_at          | updated_at          | deleted_at |
+------------------+-----------+----------+-----------+-----------+------------+----------+------------+---------------+-------------+--------------+---------------+------------+-------------+-------+-----------+---------------+---------------------+---------------------+------------+
| jr1frr********** | person    | manual   | angel     | Angel     | NULL       |          |            | NULL          | NULL        | NULL         | NULL          |         60 |          60 | NULL  | NULL      | NULL          | 2021-10-23 15:16:19 | 2021-10-23 15:16:19 | NULL       |
| jr1frr********** | person    | manual   | c*******  | C*******  | NULL       |          |            | NULL          | NULL        | NULL         | NULL          |         40 |          40 | NULL  | NULL      | NULL          | 2021-10-23 15:16:34 | 2021-10-23 15:16:34 | NULL       |
+------------------+-----------+----------+-----------+-----------+------------+----------+------------+---------------+-------------+--------------+---------------+------------+-------------+-------+-----------+---------------+---------------------+---------------------+------------+
2 rows in set (0.000 sec)
```

This is what I see in the UI:

![photoprism-error](https://user-images.githubusercontent.com/4056725/147875793-42402afc-bc6d-49b6-993c-8ff9983ea97e.PNG)

**What version you are using?**

`211215-93b26f19-Linux-x86_64`

**Any other helpful information?**

I tried to reset all the faces using the `docker-compose exec photoprism photoprism faces reset -f`, but the result was the same.

And finally, thank you for working on this amazing application and Happy New Year 😄 !

**Comment by lastzero** [comment]
Could it be that all pictures with faces are private? What database & database version are you using? Anything special about the environment (as it seems to work well for everyone else)? Did you disable the background worker or increase the interval to a very long time?

**Comment by lastzero** [comment]
If you run

```
docker-compose exec photoprism photoprism faces reset -f
```

then all faces & people will be gone. So it's expected that the result is an empty page.

**Comment by Angelmmiguel** [comment]
Thanks for the quick response @lastzero! 

> Could it be that all pictures with faces are private? 

I don't think so. When accessing `/private`, I don't see any image there.

> What database & database version are you using?

I'm using MariaDB version `10.5.12`.

> Anything special about the environment (as it seems to work well for everyone else)? 

I think my instalation is pretty standard. I'm using the official container image and MariaDB to avoid lock errors with SQLite. I will try to got through the installation docs to see i I'm doing anything special on my side.

> Did you disable the background worker or increase the interval to a very long time?

I'm using the official container image and I didn't change any default setting related to background workers. Also, the server has been running for some time.

------

I was thinking about `subj_hidden`, `sujb_private` and `subj_excluded` columns. It seems to me they are flags and I'm not sure if having a `NULL` value may cause issues. I will try to play with those and see if something happens on my installation.

Thanks again!

**Comment by lastzero** [comment]
They should not be NULL from what I remember. Did you previously use a Development Preview version that could have initialized the table in a way that's not compatible with the stable version? Does it work when you manually set the columns to `0`?

**Comment by Angelmmiguel** [comment]
> Did you previously use a Development Preview version that could have initialized the table in a way that's not compatible with the stable version

AFAIK, I was using only stable versions (`YYYYMMDD` container tags). 

> Does it work when you manually set the columns to 0?

YES (kind of)! Now I can see the person in the list, but there's no thumbnail for them. 

Btw, this is the command I used:

```
UPDATE subjects SET subj_hidden = 0, subj_excluded = 0, subj_private = 0 WHERE subj_uid = "jr1frr***********";
```

And this is the result:

<img width="1010" alt="Captura de pantalla 2022-01-02 a las 15 13 26" src="https://user-images.githubusercontent.com/4056725/147878523-2e707add-0c5f-48d5-affc-e5bb92f0b1af.png">

**Comment by lastzero** [comment]
So it seems missing thumbs are somehow the problem. Did you change thumbnail settings or delete anything from the cache folder? Do you see the faces correctly in the Edit dialog > People tab?

**Comment by Angelmmiguel** [comment]
> Did you change thumbnail settings or delete anything from the cache folder?

I didn't change any configuration about the cache folder since the beginning of the installation.

> Do you see the faces correctly in the Edit dialog > People tab?

Yes! I tested it in several pictures and I can see the thumbnails of the different people correctly. 

For me, it seems the issue is at DB level. For some reason, the thumbnail is not properly attached when storing the `subject` record. I'm gonna try to set this value manually based on the thumbnails of existing people.

**Comment by lastzero** [comment]
Could you try upgrading to MariaDB 10.5.13 or 10.6.5?

**Comment by lastzero** [comment]
Also try running 

```
docker-compose exec photoprism photoprism faces audit --fix
```

**Comment by Angelmmiguel** [comment]
> Could you try upgrading to MariaDB 10.5.13 or 10.6.5?

Done! I upgraded my DB to `10.5.13`.

> Also try running

I tried this command and this was the result:

```
# photoprism faces audit --fix
INFO[2022-01-02T17:55:10+01:00] Help us make a difference and become a sponsor today! 
INFO[2022-01-02T17:55:10+01:00] Visit https://docs.photoprism.app/funding/ to learn more. 
INFO[2022-01-02T17:55:16+01:00] found 36 subjects                            
INFO[2022-01-02T17:55:16+01:00] found no invalid marker subjects             
INFO[2022-01-02T17:55:16+01:00] found no invalid marker faces                
INFO[2022-01-02T17:55:16+01:00] found no ambiguous subjects                  
INFO[2022-01-02T17:55:16+01:00] found no orphan face clusters                
INFO[2022-01-02T17:55:16+01:00] found no orphan people                       
INFO[2022-01-02T17:55:16+01:00] completed in 5.332241584s                    
INFO[2022-01-02T17:55:16+01:00] closed database connection      
```

However, the `/people` page still doesn't load the thumbnails. Note that if I access to `/people/new` page I see all thumbnails. I tried to tag a completely new person and the result was the same:

- No thumbnail
- `subj_hidden`, `subj_excluded` and `subj_private` are `NULL` values.

I tried to populate the `thumb` column using the hash of the image in the cache folder. However, it didn't work neither. I guess I'm missing something from the implementation POV.

**Comment by Angelmmiguel** [comment]
Also, I noticed the `person:X` filter works for all the 36 people I tagged. It doesn't matter they appear or not in the `/people` page, the filter works properly

**Comment by lastzero** [comment]
Does reindexing do the trick? Anything in the error logs?

**Comment by Angelmmiguel** [comment]
So, I tried to reindex everything using:

```
docker-compose exec photoprism photoprism faces index
```

The reindex process worked fine and I didn't see any error. Just some warnings about timestamps.

However, the people section is still missing the thumbnail and the new people I'm tagging. I will continue checking it tomorrow in case I can find something interesting to share.

**Comment by lastzero** [comment]
It's possible that very small face crops and images of low-quality (low resolution, no metadata) won't be used for people covers, so no usable thumb is available and the person is not shown. Could that be the case?

**Comment by lastzero** [comment]
In that case, it helps to also match a photo of the same person with higher quality.

**Comment by Angelmmiguel** [comment]
Mmm I don't think that's the issue. My library includes some low-res images, but also some high-res pictures and even DSLR camera pictures.

It's true that the indexation start from the beginning of the library, so most likely the first analyzed images were the low-res ones. However, this doesn't cover the issue related to the `NULL` values on the columns 🤔.

**Comment by lastzero** [comment]
This is really strange... Let us know if you find out what is causing this. If the query is broken, it shouldn't work for anyone. So it's probably something more specific, like an edge case that we didn't take into account.

**Comment by Angelmmiguel** [comment]
Okey! Thank you very much for your help on this @lastzero 😄. Now, I will try to check the code and debug it on my side. If I find something interesting, I will share with you. It seems to be an edge case since I'm using the latest version and it seems to work for everyone.

Setting the flag columns to `0` did the trick. Now I can see the names, so the thumbnails is something more aesthetic than functional.

**Comment by lastzero** [comment]
From what I remember, there's a query that hides people for which no more images exist. Just like for labels, months, etc
